streaming/java/streaming-api/src/main/java/org/ray/streaming/jobgraph/JobGraphBuilder.java

```diff
@@ -41,12 +45,19 @@ public JobGraph build() {
   }
 
   private void processStream(Stream stream) {
+    while (stream.isReferenceStream()) {
+      // Reference stream and referenced stream are the same logical stream, both refers to the
+      // same data flow transformation. We should skip reference stream to avoid applying same
+      // transformation multiple times.
+      LOG.debug("Skip reference stream {} of id {}", stream, stream.getId());
+      stream = stream.getReferencedStream();
```

Comment:
It will take big effort to read "referenceStream" and "referencedStream". How about using stream.getOriginalStream() and replace the ReferenceStream with ProxyStream?

Reply:
Good suggestion

Reply:
After changing to `ProxyStream`, those code read more easily, thanks for this very much